opm/input/eclipse/Schedule/Well/WellKeywordHandlers.cpp

```diff
@@ -916,27 +930,41 @@ void handleWLIST(HandlerContext& handlerContext)
         if (name[0] != '*')
             throw std::invalid_argument("The list name in WLIST must start with a '*'");
 
-        if (action == "NEW") {
+        switch (action) {
+        case WellListAction::New: {
             new_wlm.newList(name, wells);
+            break;
         }
-
-        if (!new_wlm.hasList(name))
-            throw std::invalid_argument("Invalid well list: " + name);
-
-        if (action == "MOV") {
+        case WellListAction::Add: {
+            new_wlm.addOrCreateWellList(name, wells);
+            break;
+        }
+        case WellListAction::Mov: {
             for (const auto& well : wells) {
                 new_wlm.delWell(well);
             }
+            new_wlm.addOrCreateWellList(name, wells);
+            break;
         }
-
-        if (action == "DEL") {
-            for (const auto& well : wells) {
-                new_wlm.delWListWell(well, name);
-            }
-        } else if (action != "NEW"){
-            for (const auto& well : wells) {
-                new_wlm.addWListWell(well, name);
+        case WellListAction::Del: {
+            if (new_wlm.hasList(name)) {
+                for (const auto& well : wells) {
+                    new_wlm.delWListWell(well, name);
+                }
+            } else {
+                const std::string msg_fmt = fmt::format("Problem with {{keyword}} in {{file}} line {{line}}\n"
+                                                        "The well list '{}' has not been defined and the operation DEL can not be applied.",
+                                                         name);
+                const auto& parseContext = handlerContext.parseContext;
+                parseContext.handleError(ParseContext::SCHEDULE_INVALID_NAME,
```

Comment:
using SCHEDULE_INVALID_NAME is a little hacky while not sure whether should introduce a new tag for this.